src/App.tsx

```diff
@@ -1,35 +1,424 @@
+  font-size: ${({ theme }) => theme.typography.label1Bold.fontSize};
+  font-weight: ${( { theme }) => theme.typography.label1Bold.fontWeight};
+  line-height: ${( {theme}) => theme.typography.label1Bold.lineHeight};
+  color: ${({ theme }) => theme.colors.blue[500]};
+  cursor: pointer;
+
+  color: ${({ theme, isSelected }) =>
+    isSelected ? theme.colors.blue[700] : theme.colors.blue[500]};
+`
+
+function RealtimeRankNav2Btn({children, onClick, isSelected,}: {children: React.ReactNode; onClick?: () => void; isSelected?: boolean}) {
+
+  return(
+    <RealtimeRankNav2BtnStyle onClick={onClick} isSelected={isSelected}>
+      {children}
+    </RealtimeRankNav2BtnStyle>
+  )
+}
+
+const RealtimeRankItemWrapper = styled.div`
+  width: 100%;
+  height: auto;
+  margin-top: 20px;
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  gap: 20px;
+  box-sizing: border-box;
+  /* border: 1px solid black; */
+`
+
+const RealtimeItem = styled.div`
+  width: 100%;
+  height: auto;
+  
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`
+const RealtimeItemImg = styled.img`
+  width: 100%;
+  height: auto;
+  max-width: 190px;
+  border-radius: 5px;
+`
+
+const RealtimeItemGrayTitle = styled.p`
+  font-size: ${({ theme}) => theme.typography['label1Bold'].fontSize};
+  
+  color: gray;
+`
+
+const RealtimeItemTitle = styled.p`
+  font-size: ${({ theme}) => theme.typography['label1Bold'].fontSize};
+  
+`
+
+const RealtimeItemPriceTitle =styled.p`
+  font-size: ${({ theme}) => theme.typography['label1Bold'].fontSize};
+  font-weight: ${({ theme}) => theme.typography['label1Bold'].fontWeight};
+  line-height: ${({ theme}) => theme.typography['label1Bold'].lineHeight};
+`
+
+function RealtimeRankItemList({ collapsed }: { collapsed: boolean }) {
+  const visibleItems = collapsed ? itemList.slice(0, 6) : itemList;
+
+  return (
+    <>
+    {visibleItems.map((item) => (
+        <RealtimeItem key={item.id}>
+          <RealtimeItemImg src={item.imageURL} alt={item.name}></RealtimeItemImg>
+          <RealtimeItemGrayTitle>{item.brandInfo.name}</RealtimeItemGrayTitle>
+          <RealtimeItemTitle>{item.brandInfo.name}</RealtimeItemTitle>
+          <RealtimeItemPriceTitle>{item.price.sellingPrice} 원</RealtimeItemPriceTitle>
+        </RealtimeItem>
+      ))}
+    </>
+  );
+}
+
+const ExtraBtnWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;  
+`
+
+const ExtraBtn = styled.button`
+  width: 500px;
+  height: 40px;
+  margin-top: 10px;
+  background-color: ${({ theme }) => theme.colors.background.default};
+  border: 1px solid ${({ theme}) => theme.colors.gray[300]};
+  border-radius: 5px;
+  font-size: ${({ theme }) => theme.typography['label1Bold'].fontSize};
+  cursor: pointer;
+`
+
+
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [selected, setSelected] = useState('');
+  const [selected2, setSelected2] = useState('');
```

Comment:
selected와 selected2는 각각 어떤 필터링을 수행하는 상태인지 판단하기 어려운 네이밍인거 같아요! 필터링의 역할을 보시고 바로 이해할 수 있는 네이밍을 선택해주시면 좋을거 같아요!
타입 추론이 될 수 있도록, 타입을 명시해주면 좋을거 같아요!

```suggestion
type RankType = 'ALL' | 'FEMALE' | 'MALE' | 'TEEN';
type SubRankType = 'WANT' | 'MANY' | 'WISH';
  const [rankType, setSelectedRankType] = useState<RankType|null>(null);
  const [subRankType, setSelectedSubRankType] = useState<SubRankType|null>(null);
```

Reply:
넵, 현재 step2 코드 리팩토링을 하고있습니다 버튼이 focus된 상태값을 저장하고 변경하는 변수, 함수들의 식별자들의 이름을 변경하였습니다.